doc/source/tutorial/optimize.rst

```diff
@@ -440,89 +440,225 @@ example using the Rosenbrock function follows:
 Constrained minimization of multivariate scalar functions (:func:`minimize`)
 ----------------------------------------------------------------------------
 
-The :func:`minimize` function also provides an interface to several
-constrained minimization algorithm. As an example, the Sequential Least
-SQuares Programming optimization algorithm (SLSQP) will be considered here.
-This algorithm allows to deal with constrained minimization problems of the
-form:
+The :func:`minimize` function provides algorithms for constrained minimization,
+namely ``'trust-constr'`` ,  ``'SLSQP'`` and ``'COBLYA'``. They require the constraints
+to be defined using slightly different structures. The method ``'trust-constr'`` requires
+the  constraints to be defined as a sequence of objects :func:`LinearConstraint` and
+:func:`NonlinearConstraint`. Methods ``'SLSQP'`` and ``'COBLYA'``, on the other hand,
+require constraints to be defined  as a sequence of dictionaries, with keys
+``type``, ``fun`` and ``jac``.
+
+As an example let us consider the constrained minimization of the Rosenbrock function:
 
 .. math::
    :nowrap:
 
-     \begin{eqnarray*} \min F(x) \\ \text{subject to } & C_j(X) =  0  ,  &j = 1,...,\text{MEQ}\\
-            & C_j(x) \geq 0  ,  &j = \text{MEQ}+1,...,M\\
-           &  XL  \leq x \leq XU , &I = 1,...,N. \end{eqnarray*}
+     \begin{eqnarray*} \min_{x_0, x_1} & ~~100\left(x_{0}-x_{1}^{2}\right)^{2}+\left(1-x_{0}\right)^{2} &\\
+                     \text{subject to: } & x_0 + 2 x_1 \leq 1 & \\
+		                         & x_0^2 + x_1 \leq 1  & \\
+		                         & x_0^2 - x_1 \leq 1  & \\
+					 & 2 x_0 + x_1 = 1 & \\
+					 & 0 \leq  x_0  \leq 1 & \\
+					 & -0.5 \leq  x_1  \leq 2.0. & \end{eqnarray*}
 
+This optimization problem has the unique solution :math:`[x_0, x_1] = [0.4149,~ 0.1701]`,
+for which only the first and fourth constraints are active.
+
+
+Trust-Region Constrained Algorithm (``method='trust-constr'``)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-As an example, let us consider the problem of maximizing the function:
+The trust-region constrained method deals with constrained minimization problems of the form:
 
 .. math::
+   :nowrap:
+
+     \begin{eqnarray*} \min_x & f(x) & \\
+          \text{subject to: } & ~~~\text{n-lb}_j  \leq c_j(x) \leq \text{n-ub}_j, & ~~~ j = 1,...,M\\
+           &  \text{lb}_i  \leq x_i \leq \text{ub}_i, & ~~~i = 1,...,N. \end{eqnarray*}
+
+When :math:`\text{n-lb}_j = \text{n-ub}_j`, the method reads the constraint as an
+equality constraint and deals with it accordingly. Besides that, one-sided constraint
+may be specified by setting :math:`\text{n-lb}_j` or :math:`\text{n-lb}_j` to ``np.inf`` with
```

Comment:
I think it should be "can", not "may" in all similar cases. Don't want to go into details (and I'm not sure), perhaps someone with a better knowledge of English can confirm.

Reply:
It is ok for me. I will change everything to "can"